### ISSUE TITLE
Views become black and inaccesible on tty switch in second monitor

### ISSUE THREAD

**Issue by Luminarys** [swaywm/sway#141]
I have a dual monitor setup, and when using sway if I switch to another tty, on switching back, all views on the first monitor distort, move to the background, and become black. If I create a new view in that monitor, the views are no longer black, but they are still in the background and inaccesible. Based on my log, I believe that this is because wlc is somehow temporarily reactivating one of my disabled monitors before deactivating it again. The issue is probably caused by wlc, but it also may in part be an issue with sway.

Log:

```
Handling command 'output eDP-1 disable'
Configured output eDP-1 to -1 x -1 @ -1, -1
Handling command 'output VGA-1 res 1280x1024 pos 0,0'
Configured output VGA-1 to 1280 x 1024 @ 0, 0
Handling command 'output HDMI-A-1 res 1920x1080 pos 1280,0'
Configured output HDMI-A-1 to 1920 x 1080 @ 1280, 0
[wlc] libinput: resume
[wlc] Activating tty
[wlc] MODE: (0) 1600x900 *
[wlc] MODE: (1) 1280x1024 !
[wlc] MODE: (1) 1280x1024 
[wlc] MODE: (1) 1152x864 
[wlc] MODE: (1) 1024x768 
[wlc] MODE: (1) 1024x768 
[wlc] MODE: (1) 1024x768 
[wlc] MODE: (1) 832x624 
[wlc] MODE: (1) 800x600 
[wlc] MODE: (1) 800x600 
[wlc] MODE: (1) 800x600 
[wlc] MODE: (1) 800x600 
[wlc] MODE: (1) 848x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 720x400 
[wlc] MODE: (2) 1920x1080 !
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1680x1050 
[wlc] MODE: (2) 1280x1024 
[wlc] MODE: (2) 1280x1024 
[wlc] MODE: (2) 1440x900 
[wlc] MODE: (2) 1280x960 
[wlc] MODE: (2) 1280x800 
[wlc] MODE: (2) 1152x864 
[wlc] MODE: (2) 1280x720 
[wlc] MODE: (2) 1280x720 
[wlc] MODE: (2) 1280x720 
[wlc] MODE: (2) 1440x576 
[wlc] MODE: (2) 1024x768 
[wlc] MODE: (2) 1024x768 
[wlc] MODE: (2) 1024x768 
[wlc] MODE: (2) 1440x480 
[wlc] MODE: (2) 1440x480 
[wlc] MODE: (2) 832x624 
[wlc] MODE: (2) 800x600 
[wlc] MODE: (2) 800x600 
[wlc] MODE: (2) 800x600 
[wlc] MODE: (2) 800x600 
[wlc] MODE: (2) 720x576 
[wlc] MODE: (2) 720x480 
[wlc] MODE: (2) 720x480 
[wlc] MODE: (2) 640x480 
[wlc] MODE: (2) 640x480 
[wlc] MODE: (2) 640x480 
[wlc] MODE: (2) 640x480 
[wlc] MODE: (2) 720x400 
[wlc] eDP-1 Chose mode (0) 1600x900
Output 1 resolution changed to 1600 x 900
[wlc] EGL context is double buffered
[wlc] EGL version: 1.4 (DRI2)
[wlc] EGL vendor: Mesa Project
[wlc] EGL client APIs: OpenGL OpenGL_ES OpenGL_ES2 OpenGL_ES3 
[wlc] EGL context (RGB888)
[wlc] GL version: OpenGL ES 3.0 Mesa 10.6.5
[wlc] GL vendor: Intel Open Source Technology Center
[wlc] GLES2 renderer initialized
[wlc] Set new bsurface to output (1)
Added output 1:eDP-1
Matched output config for eDP-1
[wlc] Output (1) terminated...
[wlc] Released drm surface (0x2659f18)
[wlc] Removed bsurface from output (1)
[wlc] Removed output (1)
[wlc] VGA-1 Chose mode (0) 1280x1024
Output 1 resolution changed to 1280 x 1024
[wlc] EGL context is double buffered
[wlc] EGL version: 1.4 (DRI2)
[wlc] EGL vendor: Mesa Project
[wlc] EGL client APIs: OpenGL OpenGL_ES OpenGL_ES2 OpenGL_ES3 
[wlc] EGL context (RGB888)
[wlc] GL version: OpenGL ES 3.0 Mesa 10.6.5
[wlc] GL vendor: Intel Open Source Technology Center
[wlc] GLES2 renderer initialized
[wlc] Set new bsurface to output (1)
Added output 1:VGA-1
Matched output config for VGA-1
Set VGA-1 position to 0, 0
Adding 0x26c9d90 (1, 1280.000000x1024.000000) to 0x617d98 (0, 0.000000x0.000000)
Workspace: Generating new name
Got valid workspace command for target: 'web'
Workspace: Found free name web
Added workspace web for output 1
Adding 0x26be120 (2, 1280.000000x1024.000000) to 0x26c9d90 (1, 1280.000000x1024.000000)
Setting focus to 0x26c9d90:1
[wlc] Added output (1)
[wlc] HDMI-A-1 Chose mode (0) 1920x1080
Output 2 resolution changed to 1920 x 1080
[wlc] EGL context is double buffered
[wlc] EGL version: 1.4 (DRI2)
[wlc] EGL vendor: Mesa Project
[wlc] EGL client APIs: OpenGL OpenGL_ES OpenGL_ES2 OpenGL_ES3 
[wlc] EGL context (RGB888)
[wlc] GL version: OpenGL ES 3.0 Mesa 10.6.5
[wlc] GL vendor: Intel Open Source Technology Center
[wlc] GLES2 renderer initialized
[wlc] Set new bsurface to output (2)
Added output 2:HDMI-A-1
Matched output config for HDMI-A-1
Set HDMI-A-1 position to 1280, 0
Adding 0x2b9b2a0 (1, 1920.000000x1080.000000) to 0x617d98 (0, 0.000000x0.000000)
Workspace: Generating new name
Got valid workspace command for target: 'web'
Got valid workspace command for target: 'term'
Workspace: Found free name term
Added workspace term for output 2
Adding 0x2b87990 (2, 1920.000000x1080.000000) to 0x2b9b2a0 (1, 1920.000000x1080.000000)
[wlc] Added output (2)
[wlc] become active
[wlc] Xwayland started (DISPLAY :1)
[wlc] xfixes (5.0)
Compositor is ready, executing cmds in queue
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
Handling command 'exec urxvt'
Executing urxvt
handle:3 type:0 state:0 parent:0 mask:1 (x:0 y:0 w:512 h:320) title:urxvt class:urxvt appid:(null)
Adding new view 3:urxvt to container 0x26be120 2
Adding 0x2b91b40 (4, 0.000000x0.000000) to 0x26be120 (2, 1280.000000x1024.000000)
Setting focus to 0x2b91b40:3
Arranging layout for 0x26c9d90
Arranging workspace #0 at 10.000000, 10.000000
Arranging layout for 0x26be120
Arranging 0x26be120 horizontally
Calculating arrangement for 0x2b91b40:4 (will scale 1260.000000 by 1.000000)
Set view to 1260 x 1004 @ 10, 10
focus:R|(0x617d98)(p:(nil))(f:0x26c9d90)(h:-1)Type:Root|layout:NONE|w:0.000000|h:0.000000|x:0.000000|y:0.000000|vis:t|name:(null)|children:2
|focus:X|(0x26c9d90)(p:0x617d98)(f:0x26be120)(h:1)Type:Output|layout:NONE|w:1280.000000|h:1024.000000|x:0.000000|y:0.000000|vis:f|name:VGA-1|children:1
|-focus:W|(0x26be120)(p:0x26c9d90)(f:0x2b91b40)(h:-1)Type:Workspace|layout:Horiz|w:1260.000000|h:1004.000000|x:10.000000|y:10.000000|vis:f|name:web|children:1
|--focus:K|(0x2b91b40)(p:0x26be120)(f:(nil))(h:3)Type:View|layout:NONE|w:1260.000000|h:1004.000000|x:10.000000|y:10.000000|vis:t|name:urxvt|children:0
|focus:X|(0x2b9b2a0)(p:0x617d98)(f:0x2b87990)(h:2)Type:Output|layout:NONE|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:HDMI-A-1|children:1
|-focus:X|(0x2b87990)(p:0x2b9b2a0)(f:(nil))(h:-1)Type:Workspace|layout:Horiz|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:term|children:0
[wlc] libinput: suspend
[wlc] Pending bsurface set for output (1)
[wlc] Released drm surface (0x265a268)
[wlc] Removed bsurface from output (2)
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] INPUT DEVICE REMOVED
[wlc] Released drm surface (0x2659f18)
[wlc] Removed bsurface from output (1)
[wlc] Activate vt: 2
[wlc] SIGUSR1
[wlc] libinput: suspend
[wlc] Releasing tty
[wlc] deactive
[wlc] SIGUSR2
[wlc] libinput: resume
[wlc] Activating tty
[wlc] MODE: (0) 1600x900 !
[wlc] MODE: (1) 1280x1024 !
[wlc] MODE: (1) 1280x1024 
[wlc] MODE: (1) 1152x864 
[wlc] MODE: (1) 1024x768 
[wlc] MODE: (1) 1024x768 
[wlc] MODE: (1) 1024x768 
[wlc] MODE: (1) 832x624 
[wlc] MODE: (1) 800x600 
[wlc] MODE: (1) 800x600 
[wlc] MODE: (1) 800x600 
[wlc] MODE: (1) 800x600 
[wlc] MODE: (1) 848x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 640x480 
[wlc] MODE: (1) 720x400 
[wlc] MODE: (2) 1920x1080 !
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1920x1080 
[wlc] MODE: (2) 1680x1050 
[wlc] MODE: (2) 1280x1024 
[wlc] MODE: (2) 1280x1024 
[wlc] MODE: (2) 1440x900 
[wlc] MODE: (2) 1280x960 
[wlc] MODE: (2) 1280x800 
[wlc] MODE: (2) 1152x864 
[wlc] MODE: (2) 1280x720 
[wlc] MODE: (2) 1280x720 
[wlc] MODE: (2) 1280x720 
[wlc] MODE: (2) 1440x576 
[wlc] MODE: (2) 1024x768 
[wlc] MODE: (2) 1024x768 
[wlc] MODE: (2) 1024x768 
[wlc] MODE: (2) 1440x480 
[wlc] MODE: (2) 1440x480 
[wlc] MODE: (2) 832x624 
[wlc] MODE: (2) 800x600 
[wlc] MODE: (2) 800x600 
[wlc] MODE: (2) 800x600 
[wlc] MODE: (2) 800x600 
[wlc] MODE: (2) 720x576 
[wlc] MODE: (2) 720x480 
[wlc] MODE: (2) 720x480 
[wlc] MODE: (2) 640x480 
[wlc] MODE: (2) 640x480 
[wlc] MODE: (2) 640x480 
[wlc] MODE: (2) 640x480 
[wlc] MODE: (2) 720x400 
[wlc] eDP-1 Chose mode (0) 1600x900
Output 1 resolution changed to 1600 x 900
Arranging layout for 0x617d98
Arranging output at 0
Arranging layout for 0x26c9d90
Arranging workspace #0 at 10.000000, 10.000000
Arranging layout for 0x26be120
Arranging 0x26be120 horizontally
Calculating arrangement for 0x2b91b40:4 (will scale 1580.000000 by 1.253968)
Set view to 1580 x 880 @ 10, 10
focus:R|(0x617d98)(p:(nil))(f:0x26c9d90)(h:-1)Type:Root|layout:NONE|w:0.000000|h:0.000000|x:0.000000|y:0.000000|vis:t|name:(null)|children:2
|focus:X|(0x26c9d90)(p:0x617d98)(f:0x26be120)(h:1)Type:Output|layout:NONE|w:1600.000000|h:900.000000|x:0.000000|y:0.000000|vis:f|name:VGA-1|children:1
|-focus:W|(0x26be120)(p:0x26c9d90)(f:0x2b91b40)(h:-1)Type:Workspace|layout:Horiz|w:1580.000000|h:880.000000|x:10.000000|y:10.000000|vis:f|name:web|children:1
|--focus:K|(0x2b91b40)(p:0x26be120)(f:(nil))(h:3)Type:View|layout:NONE|w:1580.000000|h:880.000000|x:10.000000|y:10.000000|vis:t|name:urxvt|children:0
|focus:X|(0x2b9b2a0)(p:0x617d98)(f:0x2b87990)(h:2)Type:Output|layout:NONE|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:HDMI-A-1|children:1
|-focus:X|(0x2b87990)(p:0x2b9b2a0)(f:(nil))(h:-1)Type:Workspace|layout:Horiz|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:term|children:0
Arranging output at 1600
Arranging layout for 0x2b9b2a0
Arranging workspace #0 at 10.000000, 10.000000
Arranging layout for 0x2b87990
focus:R|(0x617d98)(p:(nil))(f:0x26c9d90)(h:-1)Type:Root|layout:NONE|w:0.000000|h:0.000000|x:0.000000|y:0.000000|vis:t|name:(null)|children:2
|focus:X|(0x26c9d90)(p:0x617d98)(f:0x26be120)(h:1)Type:Output|layout:NONE|w:1600.000000|h:900.000000|x:0.000000|y:0.000000|vis:f|name:VGA-1|children:1
|-focus:W|(0x26be120)(p:0x26c9d90)(f:0x2b91b40)(h:-1)Type:Workspace|layout:Horiz|w:1580.000000|h:880.000000|x:10.000000|y:10.000000|vis:f|name:web|children:1
|--focus:K|(0x2b91b40)(p:0x26be120)(f:(nil))(h:3)Type:View|layout:NONE|w:1580.000000|h:880.000000|x:10.000000|y:10.000000|vis:t|name:urxvt|children:0
|focus:X|(0x2b9b2a0)(p:0x617d98)(f:0x2b87990)(h:2)Type:Output|layout:NONE|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:HDMI-A-1|children:1
|-focus:X|(0x2b87990)(p:0x2b9b2a0)(f:(nil))(h:-1)Type:Workspace|layout:Horiz|w:1900.000000|h:1060.000000|x:10.000000|y:10.000000|vis:f|name:term|children:0
[wlc] EGL context is double buffered
[wlc] EGL version: 1.4 (DRI2)
[wlc] EGL vendor: Mesa Project
[wlc] EGL client APIs: OpenGL OpenGL_ES OpenGL_ES2 OpenGL_ES3 
[wlc] EGL context (RGB888)
[wlc] GL version: OpenGL ES 3.0 Mesa 10.6.5
[wlc] GL vendor: Intel Open Source Technology Center
[wlc] GLES2 renderer initialized
[wlc] Set new bsurface to output (1)
Added output 1:eDP-1
Matched output config for eDP-1
[wlc] Output (1) terminated...
OUTPUT: Destroying output '1'
Switching to workspace 0x2b87990:term
Setting focus to 0x2b87990:-1
Arranging layout for 0x2b87990
focus:R|(0x617d98)(p:(nil))(f:0x2b9b2a0)(h:-1)Type:Root|layout:NONE|w:0.000000|h:0.000000|x:0.000000|y:0.000000|vis:t|name:(null)|children:1
|focus:X|(0x2b9b2a0)(p:0x617d98)(f:0x2b87990)(h:2)Type:Output|layout:NONE|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:HDMI-A-1|children:1
|-focus:K|(0x2b87990)(p:0x2b9b2a0)(f:(nil))(h:-1)Type:Workspace|layout:Horiz|w:1900.000000|h:1060.000000|x:10.000000|y:10.000000|vis:f|name:term|children:0
[wlc] Released drm surface (0x2659f18)
[wlc] Removed bsurface from output (1)
[wlc] Removed output (1)
[wlc] VGA-1 Chose mode (0) 1280x1024
Output 1 resolution changed to 1280 x 1024
[wlc] EGL context is double buffered
[wlc] EGL version: 1.4 (DRI2)
[wlc] EGL vendor: Mesa Project
[wlc] EGL client APIs: OpenGL OpenGL_ES OpenGL_ES2 OpenGL_ES3 
[wlc] EGL context (RGB888)
[wlc] GL version: OpenGL ES 3.0 Mesa 10.6.5
[wlc] GL vendor: Intel Open Source Technology Center
[wlc] GLES2 renderer initialized
[wlc] Set new bsurface to output (1)
Added output 1:VGA-1
Matched output config for VGA-1
Set VGA-1 position to 0, 0
Adding 0x2bb6cc0 (1, 1280.000000x1024.000000) to 0x617d98 (0, 0.000000x0.000000)
Workspace: Generating new name
Got valid workspace command for target: 'web'
Workspace: Found free name web
Added workspace web for output 1
Adding 0x2a6c350 (2, 1280.000000x1024.000000) to 0x2bb6cc0 (1, 1280.000000x1024.000000)
Setting focus to 0x2bb6cc0:1
[wlc] Added output (1)
[wlc] HDMI-A-1 Chose mode (0) 1920x1080
[wlc] EGL context is double buffered
[wlc] EGL version: 1.4 (DRI2)
[wlc] EGL vendor: Mesa Project
[wlc] EGL client APIs: OpenGL OpenGL_ES OpenGL_ES2 OpenGL_ES3 
[wlc] EGL context (RGB888)
[wlc] GL version: OpenGL ES 3.0 Mesa 10.6.5
[wlc] GL vendor: Intel Open Source Technology Center
[wlc] GLES2 renderer initialized
[wlc] Set new bsurface to output (2)
Added output 2:HDMI-A-1
Matched output config for HDMI-A-1
Set HDMI-A-1 position to 1280, 0
Adding 0x26d51e0 (1, 1920.000000x1080.000000) to 0x617d98 (0, 0.000000x0.000000)
Workspace: Generating new name
Got valid workspace command for target: 'web'
Got valid workspace command for target: 'term'
Got valid workspace command for target: 'dev'
Workspace: Found free name dev
Added workspace dev for output 2
Adding 0x2af7bb0 (2, 1920.000000x1080.000000) to 0x26d51e0 (1, 1920.000000x1080.000000)
[wlc] Added output (2)
[wlc] become active
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
[wlc] INPUT DEVICE ADDED
Handling command 'exit'
[wlc] Terminating wlc...
[wlc] Terminating compositor...
[wlc] Terminating output (1)...
[wlc] Output (2) terminated...
OUTPUT: Destroying output '2'
Switching to workspace 0x2a6c350:web
Setting focus to 0x2a6c350:-1
Arranging layout for 0x2a6c350
focus:R|(0x617d98)(p:(nil))(f:0x2bb6cc0)(h:-1)Type:Root|layout:NONE|w:0.000000|h:0.000000|x:0.000000|y:0.000000|vis:t|name:(null)|children:2
|focus:X|(0x2bb6cc0)(p:0x617d98)(f:0x2a6c350)(h:1)Type:Output|layout:NONE|w:1280.000000|h:1024.000000|x:0.000000|y:0.000000|vis:f|name:VGA-1|children:1
|-focus:K|(0x2a6c350)(p:0x2bb6cc0)(f:(nil))(h:-1)Type:Workspace|layout:Horiz|w:1280.000000|h:1024.000000|x:0.000000|y:0.000000|vis:f|name:web|children:0
|focus:X|(0x26d51e0)(p:0x617d98)(f:0x2af7bb0)(h:2)Type:Output|layout:NONE|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:HDMI-A-1|children:1
|-focus:X|(0x2af7bb0)(p:0x26d51e0)(f:(nil))(h:-1)Type:Workspace|layout:Horiz|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:dev|children:0
[wlc] Released drm surface (0x265a268)
[wlc] Removed bsurface from output (2)
[wlc] Removed output (2)
[wlc] Output (1) terminated...
OUTPUT: Destroying output '1'
Switching to workspace 0x2af7bb0:dev
Setting focus to 0x2af7bb0:-1
Arranging layout for 0x2af7bb0
focus:R|(0x617d98)(p:(nil))(f:0x26d51e0)(h:-1)Type:Root|layout:NONE|w:0.000000|h:0.000000|x:0.000000|y:0.000000|vis:t|name:(null)|children:1
|focus:X|(0x26d51e0)(p:0x617d98)(f:0x2af7bb0)(h:2)Type:Output|layout:NONE|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:HDMI-A-1|children:1
|-focus:K|(0x2af7bb0)(p:0x26d51e0)(f:(nil))(h:-1)Type:Workspace|layout:Horiz|w:1920.000000|h:1080.000000|x:1280.000000|y:0.000000|vis:f|name:dev|children:0
[wlc] Released drm surface (0x2659f18)
[wlc] Removed bsurface from output (1)
[wlc] Removed output (1)
[wlc] Compositor terminated...
[wlc] Cleanup wlc
[wlc] Closed drm
Destroying window 3
Setting focus to 0x2af7bb0:-1
[wlc] Closing Xwayland
Error: Signal 11. Printing backtrace
Backtrace: ./bin/sway(error_handler+0x66) [0x407706]
Backtrace: /usr/lib/libc.so.6(+0x33680) [0x7fc9dacc0680]
Backtrace: /usr/lib/libEGL.so(+0xf00c) [0x7fc9d607500c]
Backtrace: /usr/lib/libEGL.so(+0xea07) [0x7fc9d6074a07]
Backtrace: /usr/lib/libwayland-server.so.0(+0x65dc) [0x7fc9daa815dc]
Backtrace: /usr/lib/libwayland-server.so.0(wl_map_for_each+0x4d) [0x7fc9daa86aad]
Backtrace: /usr/lib/libwayland-server.so.0(wl_client_destroy+0x78) [0x7fc9daa82218]
Backtrace: /usr/local/lib64/libwlc.so.0(wlc_xwayland_terminate+0x64) [0x7fc9db718195]
Backtrace: /usr/local/lib64/libwlc.so.0(wlc_cleanup+0x43) [0x7fc9db717157]
Backtrace: /usr/local/lib64/libwlc.so.0(wlc_run+0xba) [0x7fc9db7174d0]
Backtrace: ./bin/sway(main+0x336) [0x40c416]
Backtrace: /usr/lib/libc.so.6(__libc_start_main+0xf0) [0x7fc9dacad610]
Backtrace: ./bin/sway(_start+0x29) [0x403e89]
```

**Comment by SirCmpwn** [comment]
Report upstream to wlc.

**Comment by SirCmpwn** [comment]
<taiyu> github wont let me post.... that tty switching is our fault, not wlcs. we dont handle destroyed outputs properly, so the views are lost.

**Comment by Luminarys** [comment]
After the recent updates, sway crashes when I do the same thing. Here is the error and bt:
```
sway: /home/luminarys/Installs/wlc/src/compositor/output.c:532: wlc_output_set_information: Assertion `output->active.mode != (0x7fffffff * 2U + 1U) && "output should have at least one current mode!"' failed.
Error: Signal 6. Printing backtrace
Backtrace: ./bin/sway(error_handler+0x66) [0x407706]
Backtrace: /usr/lib/libc.so.6(+0x33680) [0x7fec1e833680]
Backtrace: /usr/lib/libc.so.6(gsignal+0x38) [0x7fec1e8335f8]
Backtrace: /usr/lib/libc.so.6(abort+0x16a) [0x7fec1e834a7a]
Backtrace: /usr/lib/libc.so.6(+0x2c417) [0x7fec1e82c417]
Backtrace: /usr/lib/libc.so.6(+0x2c4c2) [0x7fec1e82c4c2]
Backtrace: /usr/local/lib64/libwlc.so.0(wlc_output_set_information+0x167) [0x7fec1f270581]
Backtrace: /usr/local/lib64/libwlc.so.0(+0xe547) [0x7fec1f26d547]
Backtrace: /usr/local/lib64/libwlc.so.0(+0xe80f) [0x7fec1f26d80f]
Backtrace: /usr/local/lib64/libwlc.so.0(+0x1b40b) [0x7fec1f27a40b]
Backtrace: /usr/local/lib64/libwlc.so.0(+0x1c336) [0x7fec1f27b336]
Backtrace: /usr/local/lib64/libwlc.so.0(+0x1ce50) [0x7fec1f27be50]
Backtrace: /usr/local/lib64/libwlc.so.0(wlc_backend_update_outputs+0x5b) [0x7fec1f27a2ae]
Backtrace: /usr/local/lib64/libwlc.so.0(+0xddc4) [0x7fec1f26cdc4]
Backtrace: /usr/local/lib64/libwlc.so.0(+0x2ab89) [0x7fec1f289b89]
Backtrace: /usr/local/lib64/libwlc.so.0(wlc_set_active+0x44) [0x7fec1f28a077]
Backtrace: /usr/local/lib64/libwlc.so.0(+0x29645) [0x7fec1f288645]
Backtrace: /usr/lib/libc.so.6(+0x33680) [0x7fec1e833680]
Backtrace: /usr/lib/libc.so.6(epoll_wait+0x13) [0x7fec1e8e9703]
Backtrace: /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch+0x43) [0x7fec1e5f7213]
```

**Comment by Cloudef** [comment]
@Luminarys That looks like regression inside wlc. Maybe post bug there as well.

**Comment by SirCmpwn** [comment]
Still a problem? I can switch between ttys without any issues.

**Comment by diggit** [comment]
With latest wlc and sway switching between ttys and sway is ok. But I get black screen when switching from X to sway directly. Then I have to kill sway through ssh. Using Archlinux.

**Comment by SirCmpwn** [comment]
I'm closing issues that are resolved by the yet-unreleased sway 1.0.
